examples/dell5k_edge_system_with_filter_iotcc.py

```diff
@@ -0,0 +1,203 @@
+
+import psutil
+
+from liota.dccs.iotcc import IotControlCenter
+from liota.entities.metrics.metric import Metric
+from liota.entities.devices.simulated_device import SimulatedDevice
+from liota.entities.edge_systems.dell5k_edge_system import Dell5KEdgeSystem
+from liota.dcc_comms.websocket_dcc_comms import WebSocketDccComms
+from liota.dccs.dcc import RegistrationFailure
+from liota.lib.utilities.filters.range_filter import RangeFilter, Type
+from liota.lib.utilities.filters.windowed_filters.windowed_range_filter import WindowedRangeFilter
+
+# getting values from conf file
+config = {}
+execfile('sampleProp.conf', config)
+
+# some standard metrics for Linux systems
+# agent classes for different IoT system
+# agent classes for different data center components
+# agent classes for different kinds of of devices, connected to the IoT System
+# we are showing here how to create a representation for a Device in IoTCC but
+# using the notion of RAM (because we have no connected devices yet)
+# agent classes for different kinds of layer 4/5 connections from agent to DCC
+# -------User defined functions for getting the next value for a metric --------
+# usage of these shown below in main
+# semantics are that on each call the function returns the next available value
+# from the device or system associated to the metric.
+
+# Filters to filter data at Sampling Functions
+# Simple filters
+cpu_pro_filter = RangeFilter(Type.CLOSED_REJECT, 5, 10)  # If no of CPU processes <=5 or >=10
+cpu_util_filter = RangeFilter(Type.AT_LEAST, None, 85)  # CPU util >= 85
+disk_usage_filter = RangeFilter(Type.AT_LEAST, None, 80)  # Disk usage >= 80%
+
+# Filters with windowing scheme
+net_usage_filter = WindowedRangeFilter(Type.AT_LEAST, None, 1000000, 30)  # Network usage >= 1Mb
+mem_free_filter = WindowedRangeFilter(Type.AT_MOST, 15, None, 60)  # Memory Free <= 15%
+
+
+# ---------------------------------------------------------------------------
+# User defined methods with RangeFilters
+
+
+def read_cpu_procs():
+    cnt = 0
+    procs = psutil.pids()
+    for i in procs[:]:
+        p = psutil.Process(i)
+        if p.status() == 'running':
+            cnt += 1
+    return cpu_pro_filter.filter(cnt)
+
+
+def read_cpu_utilization(sample_duration_sec=1):
+    return cpu_util_filter.filter(round(psutil.cpu_percent(interval=sample_duration_sec), 2))
+
+
+def read_disk_usage_stats():
+    return disk_usage_filter.filter(round(psutil.disk_usage('/').percent, 2))
+
+
+def read_network_bytes_received():
+    return net_usage_filter.filter(round(psutil.net_io_counters(pernic=False).bytes_recv, 2))
+
+
+def read_mem_free():
+    return mem_free_filter.filter(round((100 - psutil.virtual_memory().percent), 2))
+
+
+# ---------------------------------------------------------------------------
+# In this example, we demonstrate how System health and some simluated data
+# can be directed to two data center components (IoTCC and graphite) using Liota.
```

Comment:
This comment does not seem relevant, we are sending data only to IoTCC in this case. In general, check all comments.